python/plugins/processing/algs/qgis/RasterSampling.py

```diff
@@ -0,0 +1,196 @@
+            QgsProcessingParameterRasterLayer(
+                self.RASTERCOPY,
+                self.tr('Raster Layer to sample'),
+            )
+        )
+
+        columnPrefix = QgsProcessingParameterString(
+            self.COLUMN_PREFIX,
+            self.tr('Output column prefix'), 'rvalue'
+        )
+        columnPrefix.setFlags(columnPrefix.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
+        self.addParameter(columnPrefix)
+
+        self.addParameter(
+            QgsProcessingParameterFeatureSink(
+                self.OUTPUT,
+                self.tr('Sampled Points')
+            )
+        )
+
+    def processAlgorithm(self, parameters, context, feedback):
+
+        source = self.parameterAsSource(
+            parameters,
+            self.INPUT,
+            context
+        )
+
+        sampled_raster = self.parameterAsRasterLayer(
+            parameters,
+            self.RASTERCOPY,
+            context
+        )
+
+        columnPrefix = self.parameterAsString(
+            parameters,
+            self.COLUMN_PREFIX,
+            context
+        )
+
+        if source is None:
+            raise QgsProcessingException(self.invalidSourceError(parameters, self.INPUT))
+
+        source_fields = source.fields()
+        raster_fields = QgsFields()
+
+        # append field to vector as columnPrefix_bandCount
+        for b in range(sampled_raster.bandCount()):
+            raster_fields.append(QgsField(
+                columnPrefix + str('_{}'.format(b + 1)), QVariant.Double
+            )
+            )
+
+        # combine all the vector fields
+        out_fields = QgsProcessingUtils.combineFields(source_fields, raster_fields)
+
+        (sink, dest_id) = self.parameterAsSink(
+            parameters,
+            self.OUTPUT,
+            context,
+            out_fields,
+            source.wkbType(),
+            source.sourceCrs()
+        )
+
+        if sink is None:
+            raise QgsProcessingException(self.invalidSinkError(parameters, self.OUTPUT))
+
+        total = 100.0 / source.featureCount() if source.featureCount() else 0
+        features = source.getFeatures()
+
+        # create the coordinates transformation context
+        ct = QgsCoordinateTransform(source.sourceCrs(), sampled_raster.crs(), context.transformContext())
+
+        for n, i in enumerate(source.getFeatures()):
+
+            if i.geometry().isMultipart():
+                raise QgsProcessingException(self.tr('''Impossible to sample data
+                of a Multipart layer. Please use the Multipart to single part
+                algorithm to transform the layer.'''))
+
+            # get the feature geometry as point
+            point = i.geometry().asPoint()
+
+            # reproject to raster crs
+            try:
+                point = ct.transform(point)
+            except QgsCsException:
+                feedback.reportError(self.tr('Could not reproject feature {} to raster CRS').format(i.id()))
```

Comment:
Missed something here - better set all the raster attributes as null for this feature if this happens to avoid misleading results.

Reply:
something like:

```python
try:
    point = ct.transform(point)
except QgsCsException:
    attr.append(None)
    i.setAttributes(attr)
    feedback.reportError(self.tr('Could not reproject feature {} to raster CRS').format(i.id()))
```

?

Reply:
You'll need to add as many nulls as there are bands here.

Reply:
mmm ok.. but doesn't the `QgsCsException` stop the algorithm (the loop in this case)? If yes, isn't the attribute table already automatically filled with NULLs?

Reply:
It won't stop the loop, because it's been caught. So then the identify method will be called using the original point geometry, not a reprojected one.

Reply:
Not sure if the addition is what you have in mind. It adds `None` within the Exception looping on all raster bands